src/System.Private.ServiceModel/tests/Scenarios/Infrastructure/SetupValidationTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using Infrastructure.Common;
+using System;
+using Xunit;
+
+public partial class SetupValidationTests
+{
+    [Fact]
+    [OuterLoop]
+    public static void Root_Certificate_Available_From_Server()
+    {
+        // *** SETUP *** \\
+        byte[] certData = null;
+        Exception exception = null;
+
+        // *** EXECUTE *** \\
+        try
+        {
+            certData = ServiceUtilHelper.GetRootCertificateDataFromServer();
+        }
+        catch (Exception e)
+        {
+            exception = e;
+        }
+
+        // *** VALIDATE *** \\
+        Assert.True(exception == null,
+                    String.Format("The root certificate could not be obtained from the server{0}{1}",
+                                  Environment.NewLine,
+                                  exception == null ? String.Empty : exception.ToString()));
+
+        Assert.True(certData != null, "The server returned a null root certificate");
+        Assert.True(certData.Length > 0, "The server returned an empty root certificate");
```

Comment:
I think that's a great idea.  I'll look at what's involved.  It would be great not to rely on the server's response to diagnose what's wrong with the cert.